core/node/node_framework/src/implementations/layers/pk_signing_eth_client.rs

```diff
@@ -86,15 +92,41 @@ impl WiringLayer for PKSigningEthClientLayer {
                 private_key.clone(),
                 self.contracts_config.diamond_proxy_addr,
                 gas_adjuster_config.default_priority_fee_per_gas,
-                self.sl_chain_id,
+                l1_chain_id,
                 query_client,
             );
             BoundEthInterfaceForBlobsResource(Box::new(signing_client_for_blobs))
         });
+        let signing_client_for_gateway = if input.gateway_client.is_some() {
+            if self
+                .gateway_contracts_config
+                .clone()
```

Comment:
Nit: Cloning isn't necessary here (and in a couple of other similar places); it's possible to use `as_ref()` instead. Also, given that `self.gateway_contracts_config.unwrap()` is used below, it would be more idiomatic to have an `if let`:

```rust
if let Some(gateway_contracts) = &self.gateway_contracts_config {
    if gateway_contracts. gateway_chain_id != SLChainId(0) {
        // ...
    }
}
```